upgrade-2.x.sh

```diff
@@ -249,6 +250,212 @@ function device_type_match {
     echo "${match}"
 }
 
+# Pre update cleanup: remove some not-required files from the boot partition to clear some space
+function pre_update_pi_bootfiles_removal {
+    local boot_files_for_removal=('start_db.elf' 'fixup_db.dat')
+    for f in "${boot_files_for_removal[@]}"; do
+        echo "Removing $f from boot partition"
+        rm -f "/mnt/boot/$f"
+    done
+    sync
+}
+
+function pre_update_fix_bootfiles_hook {
+    log "Applying bootfiles hostapp-hook fix"
+    local bootfiles_temp
+    bootfiles_temp=$(mktemp)
+    curl -f -s -L -o "$bootfiles_temp" https://raw.githubusercontent.com/resin-os/resinhup/77401f3ecdeddaac843b26827f0a44d3b044efdd/upgrade-patches/0-bootfiles || log ERROR "Couldn't download fixed '0-bootfiles', aborting."
+    chmod 755 "$bootfiles_temp"
+    mount --bind "$bootfiles_temp"  /etc/hostapp-update-hooks.d/0-bootfiles
+}
+
+function hostapp_based_update {
+    local update_package=$1
+    local storage_driver
+    local sysroot="/mnt/sysroot/inactive"
+
+    case ${SLUG} in
+        raspberry*)
+            log "Running pre-update fixes for ${SLUG}"
+            pre_update_pi_bootfiles_removal
+            if ! version_gt "${VERSION_ID}" "2.7.6" ; then
+                pre_update_fix_bootfiles_hook
+            fi
+            ;;
+        *)
+            log "No device-specific pre-update fix for ${SLUG}"
+    esac
+
+    if ! [ -S "/var/run/docker-host.sock" ]; then
+        ## Happens on devices booting after a regular HUP update onto a hostapps enabled resinOS
+        log "Do not have docker-host running"
+        log "Clean inactive partition"
+        rm -rf "${sysroot:?}/"*
+        log "Getting storage driver info"
+        storage_driver=$(cat /boot/storage-driver)
+        log "Starting docker-host with ${storage_driver} storage driver"
+        dockerd --log-driver=journald -s="${storage_driver}" --data-root="${sysroot}/docker" -H unix:///var/run/docker-host.sock --pidfile=/var/run/docker-host.pid --exec-root=/var/run/docker-host --bip 10.114.101.1/24 --fixed-cidr=10.114.101.128/25 --iptables=false &
+        until DOCKER_HOST="unix:///var/run/docker-host.sock" docker ps &> /dev/null; do sleep 0.2; done
```

Comment:
We should try this with a timeout otherwise if host docker can't start we will end up blocked.

Reply:
@agherzan check if the changes are good for you.

I've tested all 3 of such calls, both when everything worked, and when I changed the script to not start up docker and thus see that it correctly failed/timed-out.
Any feedback's appreciated.